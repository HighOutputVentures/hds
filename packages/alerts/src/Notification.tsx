import {
  chakra,
  Icon,
  SystemStyleObject,
  ToastPosition,
  useToast,
} from '@chakra-ui/react';
import { ReactNode } from 'react';
import CloseIcon from './CloseIcon';

type Size = 'sm' | 'md';

/** @deprecated */
export type NotificationProps = {
  size?: Size;
  icon?: JSX.Element;
  title?: ReactNode;
  description?: ReactNode;
  children?: ReactNode;
  okayButton?: string | boolean;
  closeButton?: string | boolean;
  onOkay?(): void;
  onClose?(): void;

  __testId?: string;
  __iconTestId?: string;
  __titleTestId?: string;
  __descriptionTestId?: string;
  __childrenTestId?: string;
  __okayBtnTestId?: string;
  __closeBtnTestId?: string;
};

/** @deprecated */
export function Notification({
  icon,
  title,
  description,
  okayButton,
  closeButton,
  onOkay,
  onClose,
  children,

  __testId = 'hds.notification',
  __iconTestId = 'hds.notification.icon',
  __titleTestId = 'hds.notification.title',
  __descriptionTestId = 'hds.notification.description',
  __childrenTestId = 'hds.notification.children',
  __okayBtnTestId = 'hds.notification.okay-btn',
  __closeBtnTestId = 'hds.notification.close-btn',

  ...props
}: NotificationProps & SystemStyleObject) {
  return (
    <chakra.div
      sx={{
        paddingX: '12px',
        paddingY: '16px',
        display: 'flex',
        gap: '16px',
        alignItems: 'start',
        border: '1px solid',
        borderColor: 'Gray.100',
        rounded: '8px',
        width: 'full',
        maxWidth: '400px',
        bgColor: 'white',
        boxShadow:
          '0px 12px 16px -4px rgba(16, 24, 40, 0.08), 0px 4px 6px -2px rgba(16, 24, 40, 0.03)',
        ...props,
      }}
      data-testid={__testId}
    >
      {icon && <chakra.div data-testid={__iconTestId}>{icon}</chakra.div>}

      <chakra.div
        sx={{
          flexGrow: 1,
        }}
      >
        {!!title && (
          <chakra.div
            sx={{
              flexGrow: 1,
              fontSize: '14px',
              fontWeight: 'semibold',
              lineHeight: '14px',
              letterSpacing: '0.02em',
            }}
            data-testid={__titleTestId}
          >
            {title}
          </chakra.div>
        )}

        {!!description && (
          <chakra.div
            sx={{
              flexGrow: 1,
              marginTop: '4px',
              fontSize: '14px',
              lineHeight: '20px',
              letterSpacing: '0.02em',
              color: 'neutrals.600',
            }}
            data-testid={__descriptionTestId}
          >
            {description}
          </chakra.div>
        )}

        {!!children && (
          <chakra.div mt="16px" data-testid={__childrenTestId}>
            {children}
          </chakra.div>
        )}

        {!(!okayButton && !closeButton) && (
          <chakra.div
            sx={{
              marginTop: '16px',
              display: 'flex',
              gap: '12px',
            }}
          >
            {closeButton && (
              <chakra.button
                onClick={onClose}
                sx={{
                  fontSize: '16px',
                  lineHeight: '20px',
                  fontWeight: 'medium',
                  color: 'neutrals.600',
                }}
                data-testid={__closeBtnTestId}
              >
                {typeof closeButton === 'string' ? closeButton : 'Dismiss'}
              </chakra.button>
            )}

            {okayButton && (
              <chakra.button
                onClick={onOkay}
                sx={{
                  fontSize: '16px',
                  lineHeight: '20px',
                  fontWeight: 'medium',
                  color: 'brand.primary.700',
                }}
                data-testid={__okayBtnTestId}
              >
                {typeof okayButton === 'string' ? okayButton : 'View changes'}
              </chakra.button>
            )}
          </chakra.div>
        )}
      </chakra.div>

      <chakra.button
        onClick={onClose}
        sx={{
          display: 'flex',
          rounded: '4px',
          alignItems: 'center',
          justifyContent: 'center',
          color: 'neutrals.500',
          transition: 'all 300ms ease-in-out',
          _hover: {
            bgColor: 'neutrals.100',
            color: 'neutrals.600',
          },
        }}
        data-testid="hds.notification.btn.x"
      >
        <Icon as={CloseIcon} w="20px" h="20px" />
      </chakra.button>
    </chakra.div>
  );
}

type ToastNotificationArgs = NotificationProps & {
  position?: ToastPosition;
};

export function useNotification() {
  const toast = useToast();

  return function notify({
    position = 'bottom-right',
    onOkay,
    onClose,
    ...others
  }: ToastNotificationArgs = {}) {
    return toast({
      position,
      duration: 5000,
      render(args) {
        return (
          <Notification
            {...others}
            onOkay={() => {
              onOkay?.();
              args.onClose();
            }}
            onClose={() => {
              onClose?.();
              args.onClose();
            }}
          />
        );
      },
    });
  };
}
