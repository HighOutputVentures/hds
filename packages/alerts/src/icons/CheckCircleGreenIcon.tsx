import { Icon, IconProps } from '@chakra-ui/react';
import React, { FC } from 'react';

const CheckCircleGreenIcon: FC<Omit<IconProps, 'children' | 'css'>> = (
  props
) => (
  <Icon width="20px" height="20px" viewBox="0 0 20 20" fill="none" {...props}>
    <path
      d="M6.25008 9.99996L8.75008 12.5L13.7501 7.49996M18.3334 9.99996C18.3334 14.6023 14.6025 18.3333 10.0001 18.3333C5.39771 18.3333 1.66675 14.6023 1.66675 9.99996C1.66675 5.39759 5.39771 1.66663 10.0001 1.66663C14.6025 1.66663 18.3334 5.39759 18.3334 9.99996Z"
      stroke="#139E19"
      stroke-width="1.66667"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </Icon>
);

export default CheckCircleGreenIcon;
