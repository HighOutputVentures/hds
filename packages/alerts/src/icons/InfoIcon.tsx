import { Icon, IconProps } from '@chakra-ui/react';
import React, { FC } from 'react';

const InfoIcon: FC<Omit<IconProps, 'children' | 'css'>> = (props) => (
  <Icon width="20px" height="20px" viewBox="0 0 20 20" fill="none" {...props}>
    <path
      d="M10.0001 13.3334V10M10.0001 6.66669H10.0084M18.3334 10C18.3334 14.6024 14.6025 18.3334 10.0001 18.3334C5.39771 18.3334 1.66675 14.6024 1.66675 10C1.66675 5.39765 5.39771 1.66669 10.0001 1.66669C14.6025 1.66669 18.3334 5.39765 18.3334 10Z"
      stroke="#4A3880"
      stroke-width="1.66667"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </Icon>
);

export default InfoIcon;
