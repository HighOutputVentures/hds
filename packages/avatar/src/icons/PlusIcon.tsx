import * as React from "react";

export default function PlusIcon(props: React.ComponentProps<"svg">) {
  return (
    <svg viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
      <path
        d="M7.99992 3.33331V12.6666M3.33325 7.99998H12.6666"
        stroke="currentColor"
        strokeWidth="1.33333"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
}
