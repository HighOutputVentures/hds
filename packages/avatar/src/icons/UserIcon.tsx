import * as React from "react";

export default function UserIcon(props: React.ComponentProps<"svg">) {
  return (
    <svg
      width="32"
      height="32"
      viewBox="0 0 32 32"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M26.6666 28C26.6666 26.1392 26.6666 25.2089 26.4369 24.4518C25.9199 22.7473 24.586 21.4134 22.8814 20.8963C22.1244 20.6667 21.194 20.6667 19.3332 20.6667H12.6666C10.8058 20.6667 9.87546 20.6667 9.1184 20.8963C7.41386 21.4134 6.07997 22.7473 5.5629 24.4518C5.33325 25.2089 5.33325 26.1392 5.33325 28M21.9999 10C21.9999 13.3137 19.3136 16 15.9999 16C12.6862 16 9.99992 13.3137 9.99992 10C9.99992 6.68629 12.6862 4 15.9999 4C19.3136 4 21.9999 6.68629 21.9999 10Z"
        stroke="currentColor"
        strokeWidth="2.66667"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
}
