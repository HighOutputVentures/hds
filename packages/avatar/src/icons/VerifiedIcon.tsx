import * as React from 'react';

export default function VerifiedIcon(props: React.ComponentProps<'svg'>) {
  return (
    <svg viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
      <g clipPath="url(#clip0_3302_419216)">
        <path
          d="M15.4442 3.54197C15.6342 4.00136 15.9988 4.36652 16.4579 4.55716L18.0678 5.22398C18.5272 5.41428 18.8922 5.7793 19.0825 6.23873C19.2728 6.69815 19.2728 7.21436 19.0825 7.67379L18.4162 9.28249C18.2258 9.74212 18.2255 10.2589 18.4168 10.7183L19.082 12.3265C19.1763 12.554 19.2249 12.7979 19.2249 13.0443C19.225 13.2906 19.1765 13.5345 19.0822 13.7621C18.9879 13.9897 18.8497 14.1965 18.6755 14.3706C18.5013 14.5448 18.2945 14.6829 18.0669 14.7771L16.4582 15.4434C15.9988 15.6334 15.6336 15.998 15.443 16.4571L14.7762 18.067C14.5859 18.5264 14.2209 18.8914 13.7614 19.0817C13.302 19.272 12.7858 19.272 12.3264 19.0817L10.7177 18.4154C10.2582 18.2255 9.7422 18.2259 9.28304 18.4164L7.67318 19.0823C7.21401 19.2722 6.69825 19.272 6.2392 19.0819C5.78014 18.8917 5.41533 18.5271 5.2249 18.0682L4.55788 16.4578C4.36793 15.9985 4.00332 15.6333 3.54421 15.4427L1.93435 14.7758C1.47513 14.5856 1.11022 14.2208 0.919849 13.7617C0.729477 13.3025 0.729219 12.7865 0.919131 12.3272L1.58548 10.7185C1.77531 10.259 1.77492 9.74301 1.5844 9.28386L0.91901 7.67284C0.824671 7.4453 0.776094 7.20139 0.776054 6.95506C0.776015 6.70873 0.824513 6.46481 0.91878 6.23723C1.01305 6.00965 1.15123 5.80287 1.32544 5.62872C1.49965 5.45457 1.70647 5.31645 1.93408 5.22225L3.54278 4.55591C4.00178 4.36612 4.36671 4.00198 4.55749 3.5434L5.22432 1.93354C5.41462 1.47411 5.77964 1.1091 6.23906 0.918794C6.69849 0.728493 7.2147 0.728493 7.67413 0.918794L9.28283 1.58514C9.74227 1.77497 10.2583 1.77459 10.7174 1.58406L12.328 0.919827C12.7874 0.729633 13.3034 0.729672 13.7628 0.919935C14.2221 1.1102 14.5871 1.47511 14.7774 1.93442L15.4444 3.54476L15.4442 3.54197Z"
          fill="#0788F5"
        />
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M13.9166 7.37864C14.0502 7.16879 14.0949 6.91446 14.041 6.6716C13.9871 6.42874 13.8389 6.21724 13.6291 6.08364C13.4192 5.95004 13.1649 5.90527 12.922 5.95919C12.6792 6.0131 12.4677 6.16129 12.3341 6.37114L8.66283 12.1399L6.98283 10.0399C6.82751 9.84562 6.60138 9.72101 6.35418 9.69346C6.10699 9.66592 5.85897 9.7377 5.6647 9.89302C5.47043 10.0483 5.34582 10.2745 5.31827 10.5217C5.29073 10.7689 5.36251 11.0169 5.51783 11.2111L8.01783 14.3361C8.111 14.4528 8.23066 14.5454 8.36687 14.6065C8.50308 14.6675 8.65189 14.6951 8.80094 14.6871C8.94998 14.679 9.09494 14.6355 9.22376 14.5601C9.35258 14.4847 9.46154 14.3796 9.54158 14.2536L13.9166 7.37864Z"
          fill="white"
        />
      </g>
      <defs>
        <clipPath id="clip0_3302_419216">
          <rect width="20" height="20" fill="white" />
        </clipPath>
      </defs>
    </svg>
  );
}
