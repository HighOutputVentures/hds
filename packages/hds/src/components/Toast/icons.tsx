import * as React from 'react';

export function CheckCircleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        clipRule="evenodd"
        fillRule="evenodd"
        d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zm13.36-1.814a.75.75 0 10-1.22-.872l-3.236 4.53L9.53 12.22a.75.75 0 00-1.06 1.06l2.25 2.25a.75.75 0 001.14-.094l3.75-5.25z"
      />
    </svg>
  );
}

export function ExclamationCircleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        clipRule="evenodd"
        fillRule="evenodd"
        d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zM12 8.25a.75.75 0 01.75.75v3.75a.75.75 0 01-1.5 0V9a.75.75 0 01.75-.75zm0 8.25a.75.75 0 100-1.5.75.75 0 000 1.5z"
      />
    </svg>
  );
}

export function XMarkIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        clipRule="evenodd"
        fillRule="evenodd"
        d="M5.47 5.47a.75.75 0 011.06 0L12 10.94l5.47-5.47a.75.75 0 111.06 1.06L13.06 12l5.47 5.47a.75.75 0 11-1.06 1.06L12 13.06l-5.47 5.47a.75.75 0 01-1.06-1.06L10.94 12 5.47 6.53a.75.75 0 010-1.06z"
      />
    </svg>
  );
}

export function InformationCircleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zm8.706-1.442c1.146-.573 2.437.463 2.126 1.706l-.709 2.836.042-.02a.75.75 0 01.67 1.34l-.04.022c-1.147.573-2.438-.463-2.127-1.706l.71-2.836-.042.02a.75.75 0 11-.671-1.34l.041-.022zM12 9a.75.75 0 100-1.5.75.75 0 000 1.5z"
        fillRule="evenodd"
        clipRule="evenodd"
      />
    </svg>
  );
}

export function ExclamationTriangleIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      fill="currentColor"
      {...props}
    >
      <path
        d="M9.401 3.003c1.155-2 4.043-2 5.197 0l7.355 12.748c1.154 2-.29 4.5-2.599 4.5H4.645c-2.309 0-3.752-2.5-2.598-4.5L9.4 3.003zM12 8.25a.75.75 0 01.75.75v3.75a.75.75 0 01-1.5 0V9a.75.75 0 01.75-.75zm0 8.25a.75.75 0 100-1.5.75.75 0 000 1.5z"
        fillRule="evenodd"
        clipRule="evenodd"
      />
    </svg>
  );
}

export function SpinnerIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      {...props}
    >
      <path
        fill="currentColor"
        d="M12,1A11,11,0,1,0,23,12,11,11,0,0,0,12,1Zm0,19a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
        opacity=".25"
      ></path>
      <path
        fill="currentColor"
        d="M12,4a8,8,0,0,1,7.89,6.7A1.53,1.53,0,0,0,21.38,12h0a1.5,1.5,0,0,0,1.48-1.75,11,11,0,0,0-21.72,0A1.5,1.5,0,0,0,2.62,12h0a1.53,1.53,0,0,0,1.49-1.3A8,8,0,0,1,12,4Z"
      >
        <animateTransform
          attributeName="transform"
          dur="0.75s"
          repeatCount="indefinite"
          type="rotate"
          values="0 12 12;360 12 12"
        ></animateTransform>
      </path>
    </svg>
  );
}

export const Number123Icon = React.forwardRef<
  SVGSVGElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return (
    <svg
      ref={ref}
      xmlns="http://www.w3.org/2000/svg"
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      {...props}
    >
      <path
        fill="none"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        d="m3 10l2-2v8m4-8h3a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1h-2a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h3m4-8h2.5A1.5 1.5 0 0 1 21 9.5v1a1.5 1.5 0 0 1-1.5 1.5H18h1.5a1.5 1.5 0 0 1 1.5 1.5v1a1.5 1.5 0 0 1-1.5 1.5H17"
      ></path>
    </svg>
  );
});
