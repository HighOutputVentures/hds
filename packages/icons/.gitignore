*.log
.DS_Store
node_modules
.cache
!.vscode
storybook-static
.vercel
