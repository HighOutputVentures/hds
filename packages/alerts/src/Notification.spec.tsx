import { SearchIcon } from '@chakra-ui/icons';
import { ChakraProvider } from '@chakra-ui/react';
import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Notification from './Notification';

function Component() {
  return (
    <Notification
      alertLabel={{ label1: 'Learn More', label2: 'View Changes' }}
      alertLinks={{ link1: '#', link2: '#' }}
      supportingDetail="Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid pariatur, ipsum similique veniam."
      title="Weâ€™ve just released a new feature"
      isOpen={true}
      type="primary"
      icon={SearchIcon}
    />
  );
}

describe('Notification Component', () => {
  beforeEach(() => {
    render(
      <ChakraProvider>
        <Component />
      </ChakraProvider>
    );
  });

  it('Should render', async () => {
    const NotificationBox = await screen.findAllByTestId(
      'hds.notification.container'
    );
    expect(NotificationBox).toHaveLength(1);
  });

  it('should render title', async () => {
    const title = await screen.findAllByTestId('hds.notification.title');
    expect(title).toHaveLength(1);
  });

  it('should render supporting detail', async () => {
    const supportingDetail = await screen.findAllByTestId(
      'hds.notification.message'
    );
    expect(supportingDetail).toHaveLength(1);
  });

  it('should render notification links', async () => {
    const notificationLinks = await screen.findAllByTestId(
      'hds.notification.link'
    );
    expect(notificationLinks).toHaveLength(2);
  });

  describe('Snapshot', () => {
    it('Should match snapshot', () => {
      const component = renderer.create(<Component />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      component.unmount();
    });
  });
});
