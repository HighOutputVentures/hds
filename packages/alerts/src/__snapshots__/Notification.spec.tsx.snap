// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Component Snapshot Should match snapshot 1`] = `
<div
  className="css-1wr1rkl"
  data-testid="hds.notification.container"
>
  <svg
    className="chakra-icon chakra-icon css-kw85s7"
    focusable={false}
    viewBox="0 0 24 24"
  >
    <path
      d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
      fill="currentColor"
    />
  </svg>
  <div
    className="css-1pyc5ii"
  >
    <div
      className="chakra-stack css-ebm925"
    >
      <p
        className="chakra-text css-1xhu8vj"
        data-testid="hds.notification.title"
      >
        Weâ€™ve just released a new feature
      </p>
    </div>
    <p
      className="chakra-text css-nkyeo7"
      data-testid="hds.notification.message"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid pariatur, ipsum similique veniam.
    </p>
    <div
      className="chakra-stack css-he6z7n"
    >
      <a
        className="chakra-text css-o9hqjy"
        data-testid="hds.notification.link"
        href="#"
        target="_blank"
      >
        Learn More
      </a>
      <a
        className="chakra-text css-4g6ai3"
        data-testid="hds.notification.link"
        href="#"
        target="_blank"
      >
        <span
          className="chakra-text css-4p6fdr"
        >
          View Changes
        </span>
      </a>
    </div>
  </div>
  <div
    className="css-1t9pz9x"
  />
  <button
    aria-label="Close"
    className="css-12hfj2s"
    type="button"
  >
    <svg
      aria-hidden={true}
      className="chakra-icon css-onkibi"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;
